packages/js/product-editor/src/components/index.ts

```diff
@@ -5,3 +5,11 @@ export {
 export { WooProductFieldItem as __experimentalWooProductFieldItem } from './woo-product-field-item';
 export { WooProductSectionItem as __experimentalWooProductSectionItem } from './woo-product-section-item';
 export { WooProductTabItem as __experimentalWooProductTabItem } from './woo-product-tab-item';
+export { DetailsNameField as __experimentalDetailsNameField } from './field-details-name';
+export { DetailsFeatureField as __experimentalDetailsFeatureField } from './field-details-feature';
+export {
+	DetailsCategoriesField as __experimentalDetailsCategoriesField,
+	CategoryField as __experimental__experimentalDetailsCategoryField,
```

Comment:
This looks like a mistake... should it be `CategoryField as __experimentalCategoryField`?

Question though... do we need to export `CategoryField` at all? It isn't used anywhere.

Also, existing issue, but why are the components named like `DetailsCategoriesField` while the filename is `field-details-categories`? Why the different order of the words? It is confusing!

Reply:
> This looks like a mistake... should it be CategoryField as __experimentalCategoryField

Whoops! Addressed in e86d09c44ce1066bd01a1164bfddc7057ad7f232.

> Question though... do we need to export CategoryField at all? It isn't used anywhere.

True, and I'm glad to remove it. It's a more configurable / abstract version of the category field, so I'd thought some 3PD might find it _more_ useful than our more narrow version. I'm good either way.

> Also, existing issue, but why are the components named like DetailsCategoriesField while the filename is field-details-categories? Why the different order of the words? It is confusing!

Ha, this is probably a weird thing of mine! It feels a little more organized to have the `field`'s all listed next to eachother in the file listing so your eye can immediately see them grouped when they're listed alphabetically:

![image](https://user-images.githubusercontent.com/444632/222528182-5cc3a5ce-ed7b-4bab-b460-1a68235f616e.png)


But it also feels weird to call a component `FieldDetailsName` instead of `DetailsNameField`, so now they don't match 😅 . I suppose I can let go of my compulsions around the file-listing order, since I can see that being confusing. Does that make the most sense to you  @mattsherman ?

Reply:
It could just be me, but I found the mismatch in naming styles to introduce a level of indirection/confusion in my brain that I felt was getting in the way. So, I'd advocate keeping the naming the same between folder/file and component.

Reply:
@joelclimbsthings If we were to rename the folders/files, do you think it should be handled in this PR or a separate PR?

Reply:
I went ahead and did so in https://github.com/woocommerce/woocommerce/pull/36945/commits/7779e93127b063ced59d90e587f7b6f9a97fd110.